### PR TITLE
Hosting Command Palette: display contextual commands by default based on the current path

### PR DESCRIPTION
**Pull request by sejas**
- Related to https://github.com/Automattic/dotcom-forge/issues/4531

## Proposed Changes

* Limit the commands displayed by default and those are based on the current path

## Testing Instructions

* Run this branch locally or use calypso live
* Access `/sites`
* Press `cmd+k` on mac or `ctrl+k` on Windows
* Observe that the list of commands is limited and there is no scroll in the palette
* Type a letter , like `s`
* Observe the list of commands is displaying all the commands with `s` and the palette has scroll.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/command-pallette/use-command-pallette.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface Command {
 	subLabel?: string;
 	searchLabel: string;
 	callback: ( params: CommandCallBackParams ) => void;
-	context?: string;
+	context?: string | string[];
```
Comment (wojtekn) [inline client/components/command-pallette/use-command-pallette.tsx]: The `context` is an array of paths for all commands besides `openAccountSettings`, which still has it as string. What is the purpose of this text? Why not remove this one and keep the context array of strings in all cases?

File: client/components/command-pallette/wpcom-command-pallette.tsx [hunk 2]
```diff
@@ -65,11 +67,14 @@ export function CommandMenuGroup( {
...
 	setSelectedCommandName,
 }: CommandMenuGroupProps ) {
+	const currentPath = useSelector( ( state ) => getCurrentRoute( state ) );
 	const { commands: smpDefaultCommands } = useCommandPallette( {
 		selectedCommandName,
 		setSelectedCommandName,
 	} );
-	const commands = isContextual ? smpDefaultCommands : [];
+	const commands = isContextual
+		? smpDefaultCommands.filter( ( command ) => command.context?.includes( currentPath ) )
```
Comment (wojtekn) [inline client/components/command-pallette/wpcom-command-pallette.tsx]: Isn't a `smp` a bit cryptic? I know what it is, though 😅

File: client/components/command-pallette/wpcom-command-pallette.tsx [hunk 3]
```diff
@@ -223,24 +228,14 @@ export const WpcomCommandPalette = () => {
...
-						{ search && (
-							<CommandMenuGroup
-								search={ search }
-								close={ closeAndReset }
-								setSearch={ setSearch }
-								setPlaceholderOverride={ setPlaceholderOverride }
-								selectedCommandName={ selectedCommandName }
-								setSelectedCommandName={ setSelectedCommandName }
-							/>
-						) }
```
Comment (wojtekn) [inline client/components/command-pallette/wpcom-command-pallette.tsx:243]: Nice DRY!

**Review by kozer — APPROVED** [review]

**Comment by kozer** [comment]
It worked as expected @sejas ! Nice work! Approved!

**Review by wojtekn — COMMENTED** [review]
The code change looks good, and works as expected. I like that the command palette is much cleaner on the first encounter.

File: client/components/command-pallette/use-command-pallette.tsx (hunk 1 above)
Comment (sejas) [reply client/components/command-pallette/use-command-pallette.tsx]: My FE mind:
I thought it was a cool idea to let devs to decide if use one string or an array of strings.
Both objects have the method `.includes` which is what I use to match the paths.

My BE mind:
For consistency I removed the option of string and forced to use an array of strings.

https://github.com/Automattic/wp-calypso/pull/84491/commits/f5b33c1417172de6a7a97bfd35dae4b8f0f04f33

File: client/components/command-pallette/wpcom-command-pallette.tsx (hunk 2 above)
Comment (sejas) [reply client/components/command-pallette/wpcom-command-pallette.tsx]: I moved the filter inside the function, so I don't need to find a better name for that variable 😆 . I also think it's cleaner now.

https://github.com/Automattic/wp-calypso/pull/84491/commits/a2cd385286ac7efdd1253c257047e431e4375ade

Comment (wojtekn) [reply client/components/command-pallette/wpcom-command-pallette.tsx]: Nice cleanup!

**Review by wojtekn — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9739415

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @sejas, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Open site dashboard`
- `Open database in phpMyAdmin`
- `Manage hosting configuration`
- `Select site to open dashboard`
- `Select site to open hosting configuration`
- `open site dashboard`
- `admin`
- `wp-admin`
- `manage hosting configuration`
- `admin interface style`
- `database`
- `global edge cache`
- `hosting`
- `mysql`
- `phpmyadmin`
- `open my profile`
- `account`
- `display name`
- `gravatar`
- `open account settings`
- `profile`
- `email`
- `language`
- `payment methods`
- `manage domains`
- `dns`
- `domain mapping`
- `domain registration`
- `domain transfer`
- `email forwarding`
- `nameservers`
- `subdomains`
- `manage dns records`
- `cname`
- `mx`
- `txt`
- `php errors`
- `manage posts`
- `Import site to WordPress.com`
- `connect jetpack`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9739415).

**Pull request merged**